pulp_rpm/app/tasks/synchronizing.py

```diff
@@ -303,6 +317,33 @@ def newpkgcb(pkgId, name, arch):
             result = await downloader.run()
             metadata_pb.increment()
 
+            repomd_path = result.path
+            repomd = cr.Repomd(repomd_path)
+
+            # Caution: we are not storing when the remote was last updated, so the order of this
+            # logic must remain in this order where we first check the version number as other
+            # changes than sync could have taken place such that the date or repo version will be
+            # different from last sync
+            if (
+                optimize_sync and
+                self.repository.last_sync_remote and
+                self.remote.pk == self.repository.last_sync_remote.pk and
+                (self.repository.last_sync_repo_version ==
+                 self.repository.latest_version().number) and
+                (self.remote.pulp_last_updated <=
+                 self.repository.latest_version().pulp_created) and
+                int(repomd.revision) <= self.repository.last_sync_revision_number
+            ):
+                optimize_data = dict(message='Optimizing Sync', code='optimizing.sync')
+                with ProgressReport(**optimize_data) as optimize_pb:
+                    optimize_pb.done = 1
+                    optimize_pb.save()
+                    self.repository.last_sync_remote = self.remote
```

Comment:
This line is not needed. Line 330 guarantees that self.repository.last_sync_remote = self.remote.